networkit/cpp/scd/LocalTightnessExpansion.cpp

```diff
@@ -0,0 +1,297 @@
+
+                    if (!inShell[lv]) {
+                        inShell[lv] = true;
+                        newShellNodes.push_back(lv);
+                    } else {
+                        nodeExternalSimilarity[lv] -= scoreUv;
+                    }
+                }
+            });
+
+        for (node s : newShellNodes) {
+            localGraph.forNeighborsWithTrianglesOf(
+                localGraph.toGlobal(s), [&](node lv, edgeweight, double triangleSum) {
+                    if (!inResult[lv]) {
+                        double scoreUv = 0.0;
+                        double denom = weightedDegree[lv] * weightedDegree[s];
+                        if (denom > 0.0) {
+                            scoreUv = triangleSum / denom;
+                        }
+
+                        nodeExternalSimilarity[s] += scoreUv;
+                    }
+                });
+        }
+    };
+
+    // init community with seed set
+    for (node u : s) {
+        node lu = localGraph.ensureNodeExists(u);
+        result.insert(u);
+        inResult[lu] = true;
+        updateShell(u, lu);
+    }
+
+    // expand (main loop)
+    while (!shell.empty()) {
+        node uMax = shell.extract_top();
+        node gUMax = localGraph.toGlobal(uMax);
+#ifndef NDEBUG
+        double internalSimilarityNew = internalSimilarity + 2 * nodeInternalSimilarity[uMax];
+        double externalSimilarityNew =
+            externalSimilarity - nodeInternalSimilarity[uMax] + nodeExternalSimilarity[uMax];
+
+        {
+            double debugInternalSimilarity = .0;
+            double debugExternalSimilarity = .0;
+            for (node u : result) {
+                node lu = localGraph.ensureNodeExists(u);
+                debugInternalSimilarity += nodeInternalSimilarity[lu];
+                debugExternalSimilarity += nodeExternalSimilarity[lu];
+            }
+
+            assert(std::abs(debugInternalSimilarity - internalSimilarity) < 0.000001);
+            assert(std::abs(debugExternalSimilarity - externalSimilarity) < 0.000001);
+        }
+#endif
+
+        // if conductance decreases (i.e., improves)
+        if (externalSimilarity / internalSimilarity
+                - (alpha * nodeExternalSimilarity[uMax] - nodeInternalSimilarity[uMax])
+                      / (2 * nodeInternalSimilarity[uMax])
+            > 0) {
+            result.insert(gUMax);
+            inResult[uMax] = true;
+            updateShell(gUMax, uMax);
+
+            assert(std::abs(internalSimilarity - internalSimilarityNew) < 0.000001);
+            assert(std::abs(externalSimilarity - externalSimilarityNew) < 0.000001);
+        }
+    }
+    return result;
+}
+
+} // namespace
+
+std::set<node> LocalTightnessExpansion::expandOneCommunity(const std::set<node> &s) {
+    if (g->isWeighted()) {
+        return expandSeedSetInternal<true>(*g, s, alpha);
+    } else {
+        return expandSeedSetInternal<false>(*g, s, alpha);
+    }
+}
+
+} /* namespace NetworKit */
```

Comment:
Some debugging checks are quite long and make the code harder to follow. If it does not require too much work, I would suggest to implement them in separate functions.